scripts/o3de/o3de/repo_properties.py

```diff
@@ -305,6 +341,7 @@ def edit_repo_props(repo_path: pathlib.Path = None,
                        delete_templates: str or list = None,
                        replace_templates: pathlib.Path or list = None,
                        auto_update: str or list = None,
+                       dry_run: bool = None,
```

Comment:
`dry_run` should default to `False`, not `None`. `None` is an indeterminant state, `False` communicates the parameter better